metrics/release.go

```diff
@@ -45,34 +45,43 @@ var (
 			Name: "release_attempt_total",
 			Help: "Total number of releases processed by the operator",
 		},
-		[]string{"target_workspace"},
+		[]string{"reason", "strategy", "succeeded", "target_workspace"},
 	)
 )
 
-// RegisterCompletedRelease decrements the 'release_attempt_concurrent_total' metric and registers a new observation for
-// 'release_attempt_duration_seconds' with the elapsed time from the moment the Release attempt started (Release marked
-// as 'Running').
-func RegisterCompletedRelease(reason string, startTime, completionTime *metav1.Time, succeeded bool) {
+// RegisterCompletedRelease decrements the 'release_attempt_concurrent_total' metric, increments `release_attempt_total`
+// and registers a new observation for 'release_attempt_duration_seconds' with the elapsed time from the moment the
+// Release attempt started (Release marked as 'Running').
+func RegisterCompletedRelease(reason, strategy, targetWorkspace string, startTime, completionTime *metav1.Time, succeeded bool) {
+	labels := prometheus.Labels{
+		"reason":           reason,
+		"strategy":         strategy,
+		"succeeded":        strconv.FormatBool(succeeded),
+		"target_workspace": targetWorkspace,
+	}
+
 	ReleaseAttemptConcurrentTotal.Dec()
-	ReleaseAttemptDurationSeconds.With(
-		prometheus.Labels{"reason": reason, "succeeded": strconv.FormatBool(succeeded)},
-	).Observe(completionTime.Sub(startTime.Time).Seconds())
+	ReleaseAttemptDurationSeconds.With(labels).Observe(completionTime.Sub(startTime.Time).Seconds())
+	ReleaseAttemptTotal.With(labels).Inc()
 }
 
-// RegisterInvalidRelease increments the 'release_attempt_invalid_total' metrics.
+// RegisterInvalidRelease increments the 'release_attempt_invalid_total' and `release_attempt_total` metrics.
 func RegisterInvalidRelease(reason string) {
 	ReleaseAttemptInvalidTotal.With(prometheus.Labels{"reason": reason}).Inc()
+	ReleaseAttemptTotal.With(prometheus.Labels{
```

Comment:
`ReleaseAttemptTotal` is used in two places. Those are `RegisterCompletedRelease` and `RegisterInvalidRelease`. The reason it's there is because those are the only two places where we register metrics for a completed `Release`. When it was in `RegisterNewRelease`, we were registering in the total count a `Release` that just started so it was not possible to add a label to know if it succeeded or not.